R/proxy.R

```diff
@@ -1,28 +1,40 @@
-#' Extract underlying data
+#' Proxy and restore
 #'
 #' @description
 #'
-#' Extract the data underlying an S3 vector object, i.e. the underlying
-#' (named) atomic vector or list.
+#' `vec_proxy()` returns the memory representation of a vector. You
```

Comment:
I don't know what "memory representation" means

Reply:
For factors it's the C array of `int`, and by extension the `INTSXP` that wraps it.

I have rephrased this way:

```
#' `vec_proxy()` returns the data structure containing the values of a
#' vector. This data structure is usually the vector itself. In this
#' case the proxy is the [identity function][base::identity], which is
#' the default `vec_proxy()` method.
#'
#' You normally don't need to implement `vec_proxy()`, except in these
#' cases:
```